pattern/text/en/inflect.py

```diff
@@ -609,7 +612,12 @@ def singularize(word, pos=NOUN, custom={}):
             return word
     for x in singular_ie:
         if w.endswith(x+"s"):
-            return w
+            prefix = w[:w.find(x)]
+            return prefix+x
```

Comment:
sure, will change. i thought it captured the idea better (see how its exact same code at both sections you commented), more readable, etc. but your suggestion is def more concise and will run a bit faster